app/code/Magento/Catalog/Test/Unit/Model/CategoryLinkRepositoryTest.php

```diff
@@ -194,4 +204,66 @@ public function testDelete()
         $categoryMock->expects($this->once())->method('save');
         $this->assertTrue($this->model->delete($this->productLinkMock));
     }
+
+    public function testDeleteBySkus()
+    {
+        $categoryId = 42;
+        $productSku = "testSku";
+        $productId = 55;
+        $productPositions = [55 => 1];
+        $categoryMock = $this->createPartialMock(
+            \Magento\Catalog\Model\Category::class,
+            ['getProductsPosition', 'setPostedProducts', 'save', 'getId']
+        );
+        $this->categoryRepositoryMock->expects($this->once())->method('get')->with($categoryId)
+            ->willReturn($categoryMock);
+        $this->productResourceMock->expects($this->once())->method('getProductsIdsBySkus')
+            ->willReturn(['testSku' => $productId]);
+        $categoryMock->expects($this->once())->method('getProductsPosition')->willReturn($productPositions);
+        $categoryMock->expects($this->once())->method('setPostedProducts')->with([]);
+        $categoryMock->expects($this->once())->method('save');
+        $this->assertTrue($this->model->deleteBySkus($categoryId, [$productSku]));
+    }
+
+    /**
+     * @expectedException \Magento\Framework\Exception\InputException
+     * @expectedExceptionMessage The category doesn't contain the specified products.
+     */
+    public function testDeleteBySkusWithInputException()
```

Comment:
Please use $tthis->expectedException and $this->expectedExceptionMessage , also add description and docblock